### ISSUE TITLE
Suggest integration after pasting link

### ISSUE THREAD

**Issue by enriquesanchez** [ParabolInc/parabol#5940]
This is a child issue of #5939. It stems from #5534, where we explored various ideas and solutions. 

## Enhancement

When a user pastes a link from:
- one of the integrations that we support, and
- we detect that the integration is not currently connected

...we prompt the user to connect the integration.

## Example

When a user pastes a link to GitHub in a task card, we display a small inline banner inviting the user to connect Parabol to GitHub: `Connect Parabol to GitHub for a more streamlined experience`:

<img width="330" alt="Captura de Pantalla 2022-01-21 a la(s) 3 37 10 p m" src="https://user-images.githubusercontent.com/3276087/150603655-2bfbebfa-183c-402b-bb48-4f81712c48ae.png">

The banner is unobtrusive and easily dismissed.

Clicking on the link `Connect Parabol to GitHub` opens the OAuth/configure window in place, where the user can follow through the process of connecting. If the process is completed, the little banner is removed automatically. If, instead, the users clicks on the `X`, the banner is dismissed and we remember that choice.

## Scope
- We'll start by only implementing this feature in **task cards**. Later on we can add it to reflections and discussion threads once we've learned more.
- Let's also start by only having this flow triggered by one integration. I suggest we **start with GitHub links** but I can be convinced to start with Jira if our team prefers that.

## Related work

- See parent issue #5939 for more context.
- Issue: Extend GitHub integration to show richer content

## Acceptance criteria
- [ ] When a user pastes a link from GitHub, and the GitHub integration is not currently connected, and the user has not dismissed the prompt before, we display the inline banner described above.
- [ ] Clicking on the banner link fires up the flow to connect the integration. Ideally this happens in place, having the popup window appear on top. 
- [ ] If the users dismisses the banner instead (`X` button), the banner is removed and we remember the user's choice.

### Estimated effort:
- 24 hrs

**Comment by tianrunhe** [comment]
![](https://user-images.githubusercontent.com/1879975/151080855-9bdf06ef-7109-4722-9b84-6322c3b3fffd.png)

Is this feature also expected for hyperlinks?

**Comment by ackernaut** [comment]
@enriquesanchez 
- I like @tianrunhe ’s question: I think there may be a difference between adding a link to text and pasting a link
- If they connect I assume the link becomes the formatted version as seen in #5941

**Comment by enriquesanchez** [comment]
Thank you for the feedback @tianrunhe and @ackernaut 🙌

> Is this feature also expected for hyperlinks?

My initial reaction is that this pattern should not interfere with hyperlinks. The user is intentionally linking a specific text string, and I feel we shouldn't interfere with that. How do you feel about this?

> If they connect I assume the link becomes the formatted version as seen in Show inline embeds instead of regular links for connected integrations #5941

Correct, these two issues are closely related.

**Comment by ackernaut** [comment]
Yeah I think it’s right to start with 1 enhancement. When folks add links to specific text they are doing so with intention, no need to take over formatting there. Maybe later we could suggest adding the hyperlink as an attachment at the bottom in addition to the linked text.

**Comment by jordanh** [comment]
@mwermuth, @Dschoordsch and I were talking about how this one might be a nice one for @gitstart-parabol to try...

**Comment by gitstart-parabol** [comment]
@Dschoordsch  whats the appropriate `Authorization callback URL` while configuring a `github application` for testing? Cc @jordanh @mwermuth

**Comment by Dschoordsch** [comment]
@gitstart-parabol that's `http://localhost:3000/auth/github`
![Screenshot from 2023-01-16 11-09-47](https://user-images.githubusercontent.com/7331043/212653697-4655dd74-5b78-4d9f-a9c5-e3c5b843235e.png)

**Comment by gitstart-parabol** [comment]
> http://localhost:3000/auth/github

Thank you @Dschoordsch.
